employee-rostering-frontend/src/ui/pages/skill/SkillsPage.tsx

```diff
@@ -51,40 +47,34 @@ const mapDispatchToProps: DispatchProps = {
 
 export type Props = StateProps & DispatchProps;
 
-export class SkillsPage extends DataTable<Skill,SkillComponents, Props> {
+export class SkillsPage extends DataTable<Skill, Props> {
   constructor(props: Props) {
     super(props);
-    this.extractDataFromRow = this.extractDataFromRow.bind(this);
     this.addData = this.addData.bind(this);
     this.updateData = this.updateData.bind(this);
     this.removeData = this.removeData.bind(this);
   }
 
-  displayDataRow(data: Skill): JSX.Element[] {
-    return [<Text key={0}>{data.name}</Text>];
+  createNewDataInstance(): Skill {
+    return {
+      tenantId: this.props.tenantId,
+      name: ""
+    };
   }
 
-  createNewDataRow(dataStore: SkillComponents): JSX.Element[] {
-    dataStore.name = "";
-    return [<TextInput key={0} name="name"
-      aria-label="Name"
-      onChange={(value) => dataStore.name = value}/>];
+  displayDataRow(data: Skill): JSX.Element[] {
+    return [<Text key={0}>{data.name}</Text>];
   }
   
-  editDataRow(dataStore: SkillComponents, data: Skill): JSX.Element[] {
-    dataStore.name = data.name;
+  editDataRow(data: Skill): JSX.Element[] {
     return [<TextInput key={0} name="name"
       aria-label="Name"
       defaultValue={data.name}
-      onChange={(value) => dataStore.name = value}/>];
-  }
-  
-  isValid(editedValue: SkillComponents): boolean {
-    return editedValue.name.length > 0;
+      onChange={(value) => data.name = value}/>];
```

Comment:
This is the place where the TextInput mutates DataTables state. I think it would be better to call `this.setState(...)` here.